std-bits/table/src/main/java/org/enso/table/parsing/NumberParser.java

```diff
@@ -196,259 +26,132 @@ private static Pattern buildPattern(
    */
   public static NumberParser createIntegerParser(
       IntegerType integerTargetType,
-      boolean allowCurrency,
-      boolean allowLeadingZeros,
+      boolean allowSymbol,
+      boolean allowLeadingZeroes,
       boolean trimValues,
       String decimalPoint,
       String thousandSeparator) {
     return new NumberParser(
-        false,
+        true,
         integerTargetType,
-        allowCurrency,
-        allowLeadingZeros,
+        allowSymbol,
+        allowLeadingZeroes,
         trimValues,
-        false,
         decimalPoint,
         thousandSeparator);
   }
 
   /**
    * Creates a new decimal instance of this parser.
    *
-   * @param allowCurrency whether to allow currency symbols
-   * @param allowLeadingZeros whether to allow leading zeros
    * @param trimValues whether to trim the input values
-   * @param allowScientific whether to allow scientific notation
-   * @param decimalPoint the decimal separator to use (if null, then will be inferred)
-   * @param thousandSeparator the thousand separator to use (if null, then will be inferred)
+   * @param decimalPoint the decimal point set for the current format (if null then will be
+   *     inferred)
+   * @param thousandSeparator the thousand separator to use (if null then will be inferred)
    */
   public static NumberParser createDecimalParser(
-      boolean allowCurrency,
-      boolean allowLeadingZeros,
+      boolean allowSymbol,
+      boolean allowLeadingZeroes,
       boolean trimValues,
-      boolean allowScientific,
       String decimalPoint,
       String thousandSeparator) {
     return new NumberParser(
-        true,
-        null,
-        allowCurrency,
-        allowLeadingZeros,
-        trimValues,
-        allowScientific,
-        decimalPoint,
-        thousandSeparator);
+        false, null, allowSymbol, allowLeadingZeroes, trimValues, decimalPoint, thousandSeparator);
   }
 
+  private final IntegerType integerTargetType;
+  private final boolean isInteger;
+
+  private final FormatDetectingNumberParser parser;
+
   private NumberParser(
-      boolean allowDecimal,
+      boolean isInteger,
       IntegerType integerTargetType,
```

Comment:
This didn't really change but the `isInteger` and `integerTargetType` feel a bit redundant I think. Would be best replaced by some sum type I think.